sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -222,13 +221,13 @@ public int hashCode() {
     MAP,
     ROW;    // The field is itself a nested row.
 
-    public static final Set<TypeName> NUMERIC_TYPES = ImmutableSet.of(
+    public static final ImmutableSet<TypeName> NUMERIC_TYPES = ImmutableSet.of(
         BYTE, INT16, INT32, INT64, DECIMAL, FLOAT, DOUBLE);
-    public static final Set<TypeName> STRING_TYPES = ImmutableSet.of(STRING);
-    public static final Set<TypeName> DATE_TYPES = ImmutableSet.of(DATETIME);
-    public static final Set<TypeName> COLLECTION_TYPES = ImmutableSet.of(ARRAY);
-    public static final Set<TypeName> MAP_TYPES = ImmutableSet.of(MAP);
-    public static final Set<TypeName> COMPOSITE_TYPES = ImmutableSet.of(ROW);
+    public static final ImmutableSet<TypeName> STRING_TYPES = ImmutableSet.of(STRING);
```

Comment:
I don't think we want to make guava as part of our public API surface.

Reply:
You'll want to use annotations that suppress these as false positives.

Reply:
Good point, I missed that. We can't make them private as an alternative either, since other packages already refer to them. I'll add the @SuppressWarnings("MutableConstantField") to TypeName, which should take care of this. Thanks!